app/views/users/new.html.erb

```diff
@@ -0,0 +1,33 @@
+<!--<form accept-charset="UTF-8" action="/users" method="post">
+  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
+  <label for="username">UserName:</label>
+  <input type="text" id="username" name="user[username]"><br><br>
+  <label for="email">Email:</label>
+  <input type="email" id="email" name="user[email]"><br><br>
+  <label for="password">Password:</label>
+  <input type="password" id="password" name="user[password]"><br><br>
+  <input type="submit" value="Submit">
+</form>-->
+
+<!--<%=form_tag ('/users') do%>
+  <%=label_tag 'username', 'Username:'%><br>
+  <%=text_field_tag 'username'%><br>
+  <%=label_tag 'email', 'Email:'%><br>
+  <%=email_field_tag 'email'%><br>
+  <%=label_tag 'password', 'Password:'%><br>
+  <%=password_field_tag 'password'%><br>
+  <div><%= submit_tag 'Submit' %></div>
+<%end%>-->
+
+<%= form_for @user do |f| %>
+  <% if @user.errors.any? %>
+    <% @user.errors.full_messages.each do |e| %>
+      <%= e %><br>
+    <% end %>
+  <% end %> 
+  <br>
+  Username:<br /> <%= f.text_field :username, placeholder: "your new username" %><br /><br />
+  Email : <br /><%= f.text_field :email,  placeholder: "email@gmail.com" %><br /><br />
+  Password : <br /><%= f.password_field :password, placeholder: "password" %><br /><br />
+  <%= f.submit "Submit" %>
+<% end %>
```

Comment:
Your `form_for` tag should follow the format below for label and input tags.
```html
<%= form_for @user do |u|%> 
  <%= u.label :username %> 
  <%= u.text_field :username %>
  <%= u.label :email %> 
  <%= u.email_field :email %>
  <%= u.label :password %>
  <%= u.password_field :password %>
  <%= u.submit %>
<% end %>
```

Observe the difference. Always use the `text_field` tag for labels.